Issue with Parallel Compilation of Wannier90.

Hi here,

I have encountered an issue while trying to compile Wannier90 in parallel, as shown below:

```shell
# The OS information:
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy

# The toolchain use here is Intel oneAPI:
$ module load compiler mkl mpi
Loading compiler version 2021.1.1
Loading tbb version 2021.1.1
Loading debugger version 10.0.0
Loading compiler-rt version 2021.1.1
Loading dpl version 2021.1.1
Loading mkl version 2021.1.1
Removing compiler-rt version 2021.1.1
Loading compiler-rt version 2021.1.1
Removing tbb version 2021.1.1
Loading tbb version 2021.1.1
Loading mpi version 2021.1.1

# The problems:
werner@X10DAi:~/Public/repo/github.com/wannier-developers/wannier90.git$ make all -j44
(cd ./utility/w90pov && make )
(cd ./utility/w90vdw && make )
(cd ./src/obj && make -f ../Makefile.2 wannier)
(cd ./src/obj && make -f ../Makefile.2 libs)
make[1]: Entering directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/utility/w90pov'
(cd ./src/obj && make -f ../Makefile.2 post)
ifort -O2 -c src/general.f90 -o obj/general.o
make[1]: Entering directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/utility/w90vdw'
ifort -O2 w90vdw.f90 -o w90vdw.x
(cd ./src/obj && make -f ../Makefile.2 w90chk2chk)
(cd ./src/obj && make -f ../Makefile.2 w90spn2spn)
make[1]: Entering directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/src/obj'
mpiifort -DMPI -DMPI90 -O2 -c ../constants.F90
make[1]: Entering directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/src/obj'
mpiifort -DMPI -DMPI90 -O2 -c ../constants.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error_base.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error_base.F90
make[1]: Entering directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/src/obj'
mpiifort -DMPI -DMPI90 -O2 -c ../constants.F90
make[1]: Entering directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/src/obj'
mpiifort -DMPI -DMPI90 -O2 -c ../constants.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error_base.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error_base.F90
make[1]: Entering directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/src/obj'
mpiifort -DMPI -DMPI90 -O2 -c ../constants.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error_base.F90
mpiifort -DMPI -DMPI90 -O2 -c ../types.F90
mpiifort -DMPI -DMPI90 -O2 -c ../types.F90
mpiifort -DMPI -DMPI90 -O2 -c ../types.F90
mpiifort -DMPI -DMPI90 -O2 -c ../types.F90
mpiifort -DMPI -DMPI90 -O2 -c ../types.F90
mpiifort -DMPI -DMPI90 -O2 -c ../comms.F90
mpiifort -DMPI -DMPI90 -O2 -c ../comms.F90
mpiifort -DMPI -DMPI90 -O2 -c ../comms.F90
ifort -O2 -c src/io.f90 -o obj/io.o
mpiifort -DMPI -DMPI90 -O2 -c ../comms.F90
mpiifort -DMPI -DMPI90 -O2 -c ../comms.F90
cc  -c src/write_df3.c -o obj/write_df3.o
ifort -O2 -c src/driver.f90 -o obj/driver.o
make[1]: Leaving directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/utility/w90vdw'
ifort -O2 -c src/main.f90 -o obj/main.o
ifort -O2 -o w90pov obj/write_df3.o obj/driver.o obj/main.o obj/general.o obj/io.o
make[1]: Leaving directory '/home/werner/Public/repo/github.com/wannier-developers/wannier90.git/utility/w90pov'
mpiifort -DMPI -DMPI90 -O2 -c ../error.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error.F90
mpiifort -DMPI -DMPI90 -O2 -c ../io.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error.F90
mpiifort -DMPI -DMPI90 -O2 -c ../error.F90
mpiifort -DMPI -DMPI90 -O2 -c ../io.F90
mpiifort -DMPI -DMPI90 -O2 -c ../io.F90
mpiifort -DMPI -DMPI90 -O2 -c ../io.F90
mpiifort -DMPI -DMPI90 -O2 -c ../io.F90
mpiifort -DMPI -DMPI90 -O2 -c ../utility.F90
mpiifort -DMPI -DMPI90 -O2 -c ../ws_distance.F90
mpiifort -DMPI -DMPI90 -O2 -c ../wannier90_types.F90
mpiifort -DMPI -DMPI90 -O2 -c ../utility.F90
mpiifort -DMPI -DMPI90 -O2 -c ../ws_distance.F90
mpiifort -DMPI -DMPI90 -O2 -c ../wannier90_types.F90
../ws_distance.F90(70): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [W90_UTILITY]
    use w90_utility, only: utility_cart_to_frac, utility_frac_to_cart, utility_inverse_mat
--------^
../ws_distance.F90(175): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [W90_UTILITY]
    use w90_utility, only: utility_cart_to_frac, utility_frac_to_cart
--------^
../ws_distance.F90(70): error #6580: Name in only-list does not exist or is not accessible.   [UTILITY_CART_TO_FRAC]
    use w90_utility, only: utility_cart_to_frac, utility_frac_to_cart, utility_inverse_mat
---------------------------^
../ws_distance.F90(70): error #6580: Name in only-list does not exist or is not accessible.   [UTILITY_FRAC_TO_CART]
    use w90_utility, only: utility_cart_to_frac, utility_frac_to_cart, utility_inverse_mat
-------------------------------------------------^
../ws_distance.F90(70): error #6580: Name in only-list does not exist or is not accessible.   [UTILITY_INVERSE_MAT]
    use w90_utility, only: utility_cart_to_frac, utility_frac_to_cart, utility_inverse_mat
-----------------------------------------------------------------------^
../ws_distance.F90(175): error #6580: Name in only-list does not exist or is not accessible.   [UTILITY_CART_TO_FRAC]
    use w90_utility, only: utility_cart_to_frac, utility_frac_to_cart
---------------------------^
../ws_distance.F90(175): error #6580: Name in only-list does not exist or is not accessible.   [UTILITY_FRAC_TO_CART]
    use w90_utility, only: utility_cart_to_frac, utility_frac_to_cart
-------------------------------------------------^
compilation aborted for ../ws_distance.F90 (code 1)
make[1]: *** [../Makefile.2:142: ws_distance.o] Error 1
make[1]: *** Waiting for unfinished jobs....
mpiifort -DMPI -DMPI90 -O2 -c ../ws_distance.F90
mpiifort -DMPI -DMPI90 -O2 -c ../utility.F90
mpiifort -DMPI -DMPI90 -O2 -c ../utility.F90
mpiifort -DMPI -DMPI90 -O2 -c ../ws_distance.F90
mpiifort -DMPI -DMPI90 -O2 -c ../wannier90_types.F90
mpiifort -DMPI -DMPI90 -O2 -c ../utility.F90
mpiifort -DMPI -DMPI90 -O2 -c ../ws_distance.F90
mpiifort -DMPI -DMPI90 -O2 -c ../wannier90_types.F90
```


As you can see, during the process, I receive multiple error messages related to opening the compiled module file and checking INCLUDE paths. Here's an example of such an error: "../ws_distance.F90(70): error #7002: Error in opening the compiled module file. Check INCLUDE paths.", and so on.

Interestingly, when I do sequential compilation or I run parallel compilation multiple times, these problems will disappear. Therefore, I suspect there are issues related to the ordering of dependencies in the Makefile rules.

I would appreciate if you could provide input on how to rectify this issue. Are there certain changes or adjustments I need to make to the Makefile to ensure successful parallel compilation?

Thank you for your time and help.

Best regards,
Zhao

--- a/src/Makefile.2
+++ b/src/Makefile.2
@@ -148,7 +148,7 @@ wannierise.o: ../wannierise.F90 hamiltonian.o constants.o io.o utility.o types.o
 plot.o: ../plot.F90 constants.o io.o utility.o types.o hamiltonian.o ws_distance.o wannier90_types.o comms.o
 	 $(COMPILER) $(POSTOPTS) $(FCOPTS) -c ../plot.F90
 
-ws_distance.o: ../ws_distance.F90 constants.o io.o types.o
+ws_distance.o: ../ws_distance.F90 constants.o io.o types.o utility.o
 	 $(COMPILER) $(POSTOPTS) $(FCOPTS) -c ../ws_distance.F90
 
 transport.o: ../transport.F90 constants.o io.o types.o hamiltonian.o wannier90_types.o

